numpy/f2py/cfuncs.py

```diff
@@ -621,71 +630,122 @@
 }"""
 needs['try_pyarr_from_string'] = ['STRINGCOPYN', 'PRINTPYOBJERR', 'string']
 cfuncs['try_pyarr_from_string'] = """\
-static int try_pyarr_from_string(PyObject *obj,const string str) {
+/*
+  try_pyarr_from_string copies str[:len(obj)] to a mutable Python
+  string-like obj.
```

Comment:
```suggestion
  try_pyarr_from_string copies str[:len(obj)] to the data of an array.
```

Reply:
Indeed, the only supported Python string-like object is currently ndarray. However, this support can be extended to other array-like objects that implement, say, Array Interface protocol.